src/App.jsx

```diff
@@ -14,45 +16,99 @@ export const goodsFromServer = [
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_ALPHABETICALLY = 'Sort alphabetically';
+const SORT_LENGTH = 'Sort by length';
+
+export const App = () => {
+  const [chosenFilter, setChosenFilter] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+  let sortedGoods = [...goodsFromServer];
+
+  if (chosenFilter) {
+    switch (chosenFilter) {
+      case SORT_ALPHABETICALLY:
+        sortedGoods = sortedGoods.sort((good1, good2) => (
+          good1.localeCompare(good2)
+        ));
+        break;
+
+      case SORT_LENGTH:
+        sortedGoods = sortedGoods.sort((good1, good2) => (
+          good1.length - good2.length
+        ));
+        break;
+
+      default:
+        break;
+    }
+
+    if (isReverse) {
+      sortedGoods.reverse();
+    }
+  }
+
+  if (!chosenFilter && isReverse) {
+    sortedGoods.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const handleClickReset = () => {
+    setChosenFilter('');
+    setIsReverse(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={
+            cn('button', 'is-info',
+              { 'is-light': chosenFilter !== SORT_ALPHABETICALLY })
+          }
+          onClick={() => setChosenFilter(SORT_ALPHABETICALLY)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={
+            cn('button', 'is-success',
+              { 'is-light': chosenFilter !== SORT_LENGTH })
+          }
+          onClick={() => setChosenFilter(SORT_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={
+            cn('button', 'is-warning',
+              { 'is-light': !isReverse })
+          }
+          onClick={() => setIsReverse(!isReverse)}
+        >
+          Reverse
+        </button>
+
+        {(chosenFilter || isReverse)
+          && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={handleClickReset}
+          >
+            Reset
+          </button>
+          )}
+      </div>
+
+      <ul>
+        {sortedGoods.map((good, index) => (
+          <li data-cy="Good" key={index}>{good}</li>
```

Comment:
```suggestion
          <li data-cy="Good" key={good}>{good}</li>
```
Consider fixing tests here 
<img width="701" alt="Screenshot 2023-11-05 at 17 44 28" src="https://github.com/mate-academy/react_list-of-goods-js/assets/135232001/d2408de5-d40d-470f-bcff-ec35566eeb6a">